### PR TITLE
fix(issue-stream): Disable stack trace tooltip for performance issues [UP-146]

### PR DESCRIPTION
**Pull request by malwilley**
Added a check for `issueCategory` before rendering `<StackTraceDetails />`

However, this introduced some styling issues that I had to resolve. The stack trace applied `fontSizeLarge`, so skipping that component caused performance issues to have a smaller title. I chose to move that font size to the top-level div and override the font-size from any consumers that need a different font-size. These were:

- `<IssueLink />` (used in the activity feed tooltips)
- `<EventDetailsContent />` (used in the discover event page header)
- `<GroupHeader />` (used in the issue details page)

### CODE REVIEW

**Review by 0Calories — APPROVED** [review]
🔥

File: static/app/components/eventOrGroupTitle.spec.jsx [hunk 1]
```diff
@@ -80,4 +81,18 @@ describe('EventOrGroupTitle', function () {
...
+      <EventOrGroupTitle
+        data={{
+          ...data,
+          issueCategory: IssueCategory.PERFORMANCE,
+        }}
+        withStackTracePreview
+      />
+    );
+
+    expect(screen.queryByTestId('stacktrace-preview')).not.toBeInTheDocument();
```
Comment (evanpurkhiser) [inline static/app/components/eventOrGroupTitle.spec.jsx:96]: do we need to mouse over it or anything?

File: static/app/components/issueLink.tsx [hunk 2]
```diff
@@ -100,16 +100,19 @@ const IssueLink = ({children, orgId, issue, to, card = true}: Props) => {
...
 
-const Title = styled('h3')`
-  font-size: ${p => p.theme.fontSizeMedium};
-  margin: 0 0 ${space(0.5)};
+const Title = styled('div')`
   ${p => p.theme.overflowEllipsis};
+  margin: 0 0 ${space(0.5)};
+`;
+
+const StyledEventOrGroupTitle = styled(EventOrGroupTitle)`
```
Comment (evanpurkhiser) [inline static/app/components/issueLink.tsx:108]: Are these style changes unrelated to hiding the tooltip?

File: static/app/components/eventOrGroupTitle.spec.jsx (hunk 1 above)
Comment (malwilley) [reply static/app/components/eventOrGroupTitle.spec.jsx:96]: This test ID is on the container div that captures the mouse events so it should always be rendered

File: static/app/components/issueLink.tsx (hunk 2 above)
Comment (evanpurkhiser) [reply static/app/components/issueLink.tsx:108]: Ah sorry, I should have read!

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**